Simplenote/NSMutableAttributedString+Styling.m

```diff
@@ -16,17 +16,13 @@ - (void)processChecklistsWithColor:(UIColor *)color
 
     NSString *plainString = [self.string copy];
     NSRegularExpression *regex = allowsMultiplePerLine ? NSRegularExpression.regexForChecklistsEmbeddedAnywhere : NSRegularExpression.regexForChecklists;
-    NSArray *matches = [regex matchesInString:plainString options:0 range:plainString.fullRange];
-    NSInteger positionAdjustment = 0;
+    NSArray *matches = [[[regex matchesInString:plainString
+                                        options:0
+                                          range:plainString.fullRange] reverseObjectEnumerator] allObjects];
 
     for (NSTextCheckingResult *match in matches) {
         NSRange matchedRange = match.range;
-        if (matchedRange.location == NSNotFound) {
-            continue;
-        }
-
-        NSRange adjustedRange = NSMakeRange(matchedRange.location - positionAdjustment, matchedRange.length);
-        if (NSMaxRange(adjustedRange) > self.length) {
+        if (matchedRange.location == NSNotFound || NSMaxRange(matchedRange) > self.length) {
```

Comment:
Not sure if this is necessary since we're not using capture groups and the matched range should never be greater than self.length.  Works for a safety net if we're paranoid. :)

Reply:
Yyyyyeah let's keep it if you don't mind!! _just in case_, should not happen... buuut shouldn't hurt either!!